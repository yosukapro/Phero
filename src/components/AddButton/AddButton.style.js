import { StyleSheet } from 'react-native';

import theme from 'theme';

const Styles = StyleSheet.create({
  container: {
    width: '50%',
    alignItems: 'center',
  },
  buttonContainer: {
    width: 80,
    height: 80,
    borderRadius: 50,
  },
  image: {
    width: '30%',
    marginTop: 18,
    marginLeft: 28,
  },
  text: {
    fontFamily: theme.defaultFont,
    fontSize: 14,
    paddingTop: 15,
  },
});

export default Styles;
