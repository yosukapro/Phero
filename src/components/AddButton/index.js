import React from 'react';
import { Image, Text, TouchableOpacity, View } from 'react-native';
import PropTypes from 'prop-types';

import theme from 'theme';
import Styles from './AddButton.style';

const AddButton = ({
  background = theme.primaryColor,
  color = theme.primaryColor,
  onPress,
  text = 'AJOUTER UN CONTACT',
}) => (
  <TouchableOpacity
    style={[Styles.container]}
    onPress={onPress}
  >
    <View style={[Styles.buttonContainer, { backgroundColor: background }]}>
      <Image resizeMode='contain' source={require('assets/img/addButton.png')} style={Styles.image} />
    </View>
    <Text style={[Styles.text, { color }]}>{text}</Text>
  </TouchableOpacity>
);

AddButton.propTypes = {
  background: PropTypes.string,
  color: PropTypes.string,
  onPress: PropTypes.func,
  text: PropTypes.string,
};

export default AddButton;
