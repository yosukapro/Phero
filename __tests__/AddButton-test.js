import React from 'react';
import renderer from 'react-test-renderer';

import AddButton from '../src/components/AddButton';

test('renders correctly', () => {
  const tree = renderer.create(<AddButton />).toJSON();
  expect(tree).toMatchSnapshot();
});
