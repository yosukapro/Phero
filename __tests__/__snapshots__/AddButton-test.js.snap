// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "opacity": 1,
      "width": "50%",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "borderRadius": 50,
          "height": 80,
          "width": 80,
        },
        Object {
          "backgroundColor": "#00A3CC",
        },
      ]
    }
  >
    <Image
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../src/assets/img/addButton.png",
        }
      }
      style={
        Object {
          "marginLeft": 28,
          "marginTop": 18,
          "width": "30%",
        }
      }
    />
  </View>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "montserratRegular",
          "fontSize": 14,
          "paddingTop": 15,
        },
        Object {
          "color": "#00A3CC",
        },
      ]
    }
  >
    AJOUTER UN CONTACT
  </Text>
</View>
`;
